### ISSUE TITLE
Visual Debugger Support

### ISSUE THREAD

**Issue by githubpsyche** [quarto-dev/quarto-vscode#23]
QMD files currently don't have visual debugging support in the way Python scripts and Jupyter notebooks do. Without this feature, it's still much easier in VSCode to develop analyses using those other options.

In Interactive mode for Python scripts, You can click in the margin to the left of line numbers to set breakpoints. Then you can use Debug Cell to start a debugging session for that code cell. The debugger stops execution at breakpoints and allows you to step through code one line at a time and inspect variables (see [Debugging](https://code.visualstudio.com/docs/editor/debugging) for details.

The closest thing to this that I seem able to achieve is to add the line `import pdb; pdb.set_trace()` where I want to debug. This triggers a breakpoint where I can interrupt code flow to input commands, but I don't get to benefit from key affordances in VSCode's debugging UI.

**Comment by jjallaire** [comment]
Yes this is definitely something we'd like to do for a future version of the extension!

**Comment by GitHunter0** [comment]
This is a crucial feature in my opinion, the `Debug Cell` button in `.py` files in `VScode` is incredibly useful, I hope `Quarto` mimics that. 

I'm curious to see the blooming of Quarto project, I love the concept, it has a bright future ahead.

**Comment by jobrachem** [comment]
I agree, the ability to use VS Codes debugging is the only thing that makes me still use Jupyter Notebooks most of the time.
